src/dataflow/src/render/reduce.rs

```diff
@@ -125,6 +125,7 @@ where
                         let row = row_packer.finish_and_reuse();
                         results.push(Ok((key, row)));
                     }
+                    datums = datums_local.repurpose_allocation();
```

Comment:
Is the trait definition there (instead of a free function) mostly so this can be a `.repurpose_allocation()` rather than a `repurpose_allocation(datums_local)`? No moral objection really, but if we don't think there will ever be another implementor, it might be simpler as a boring function.